File: app/src/main/java/com/example/core/designsystem/MeridianLogo.kt
package com.example.core.designsystem

import androidx.compose.foundation.Canvas
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.size
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.StrokeCap
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

/**
 * Meridian brand mark — a wireframe globe (prime meridian + equator + curved longitude/latitude
 * lines) with a "solar noon" marker where the prime meridian crosses the rim.
 *
 * Fully theme-aware: the globe inherits [lineColor] (the Material primary) and the marker uses
 * [markerColor] (the tertiary accent), so it follows light/dark and Material You dynamic colour
 * automatically. This mirrors the adaptive launcher icon so the brand reads the same everywhere.
 */
@Composable
fun MeridianLogo(
    modifier: Modifier = Modifier,
    size: Dp = 32.dp,
    lineColor: Color = MaterialTheme.colorScheme.primary,
    markerColor: Color = MaterialTheme.colorScheme.tertiary,
) {
    Canvas(
        modifier = modifier
            .size(size)
            .semantics { contentDescription = "Meridian logo" }
    ) {
        val w = this.size.minDimension
        val center = Offset(w / 2f, w / 2f)
        val radius = w * 0.40f
        val strokeWidth = w * 0.055f
        val stroke = Stroke(width = strokeWidth, cap = StrokeCap.Round)

        // Globe rim
        drawCircle(color = lineColor, radius = radius, center = center, style = stroke)
        // Equator (horizontal diameter)
        drawLine(
            color = lineColor,
            start = Offset(center.x - radius, center.y),
            end = Offset(center.x + radius, center.y),
            strokeWidth = strokeWidth,
            cap = StrokeCap.Round
        )
        // Prime meridian (vertical diameter)
        drawLine(
            color = lineColor,
            start = Offset(center.x, center.y - radius),
            end = Offset(center.x, center.y + radius),
            strokeWidth = strokeWidth,
            cap = StrokeCap.Round
        )
        // Curved meridian (longitude ellipse)
        val meridianHalfWidth = radius * 0.42f
        drawOval(
            color = lineColor,
            topLeft = Offset(center.x - meridianHalfWidth, center.y - radius),
            size = Size(meridianHalfWidth * 2f, radius * 2f),
            style = stroke
        )
        // Curved latitude (ellipse)
        val latitudeHalfHeight = radius * 0.42f
        drawOval(
            color = lineColor,
            topLeft = Offset(center.x - radius, center.y - latitudeHalfHeight),
            size = Size(radius * 2f, latitudeHalfHeight * 2f),
            style = stroke
        )
        // Solar-noon marker where the prime meridian meets the rim
        drawCircle(
            color = markerColor,
            radius = w * 0.085f,
            center = Offset(center.x, center.y - radius)
        )
    }
}

/**
 * Logo + "Meridian" wordmark, laid out horizontally for headers and brand surfaces.
 */
@Composable
fun MeridianWordmark(
    modifier: Modifier = Modifier,
    logoSize: Dp = 28.dp,
) {
    Row(
        modifier = modifier,
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(10.dp)
    ) {
        MeridianLogo(size = logoSize)
        Text(
            text = "Meridian",
            style = MaterialTheme.typography.titleLarge,
            fontWeight = FontWeight.Black,
            color = MaterialTheme.colorScheme.onBackground
        )
    }
}

File: ios/MeridianApp/DesignSystem/MeridianLogo.swift
// MeridianLogo.swift
// Meridian — iOS 27 / Swift 6
//
// Ported from app/src/main/java/com/example/core/designsystem/MeridianLogo.kt
//
// Brand mark — a wireframe globe (prime meridian + equator + curved longitude /
// latitude lines) with a "solar noon" marker where the prime meridian crosses the
// rim — rendered with SwiftUI `Canvas`. Theme-aware via `lineColor` / `markerColor`,
// defaulting to the Meridian primary and night accents.

import SwiftUI

/// The Meridian wireframe-globe logo.
struct MeridianLogo: View {
    var size: CGFloat = 32
    var lineColor: Color = MeridianColors.primary
    /// Android uses the Material `tertiary` accent; the Meridian iOS palette uses the
    /// night accent as the tertiary-equivalent solar-noon marker tint.
    var markerColor: Color = MeridianColors.nightAccent

    var body: some View {
        Canvas { context, canvasSize in
            let w = min(canvasSize.width, canvasSize.height)
            let center = CGPoint(x: w / 2, y: w / 2)
            let radius = w * 0.40
            let strokeWidth = w * 0.055
            let stroke = StrokeStyle(lineWidth: strokeWidth, lineCap: .round)

            // Globe rim
            context.stroke(
                Path(ellipseIn: CGRect(
                    x: center.x - radius, y: center.y - radius,
                    width: radius * 2, height: radius * 2)),
                with: .color(lineColor),
                style: stroke
            )

            // Equator (horizontal diameter)
            var equator = Path()
            equator.move(to: CGPoint(x: center.x - radius, y: center.y))
            equator.addLine(to: CGPoint(x: center.x + radius, y: center.y))
            context.stroke(equator, with: .color(lineColor), style: stroke)

            // Prime meridian (vertical diameter)
            var meridian = Path()
            meridian.move(to: CGPoint(x: center.x, y: center.y - radius))
            meridian.addLine(to: CGPoint(x: center.x, y: center.y + radius))
            context.stroke(meridian, with: .color(lineColor), style: stroke)

            // Curved meridian (longitude ellipse)
            let meridianHalfWidth = radius * 0.42
            context.stroke(
                Path(ellipseIn: CGRect(
                    x: center.x - meridianHalfWidth, y: center.y - radius,
                    width: meridianHalfWidth * 2, height: radius * 2)),
                with: .color(lineColor),
                style: stroke
            )

            // Curved latitude (ellipse)
            let latitudeHalfHeight = radius * 0.42
            context.stroke(
                Path(ellipseIn: CGRect(
                    x: center.x - radius, y: center.y - latitudeHalfHeight,
                    width: radius * 2, height: latitudeHalfHeight * 2)),
                with: .color(lineColor),
                style: stroke
            )

            // Solar-noon marker where the prime meridian meets the rim
            let markerRadius = w * 0.085
            context.fill(
                Path(ellipseIn: CGRect(
                    x: center.x - markerRadius, y: center.y - radius - markerRadius,
                    width: markerRadius * 2, height: markerRadius * 2)),
                with: .color(markerColor)
            )
        }
        .frame(width: size, height: size)
        .accessibilityLabel("Meridian logo")
    }
}

/// Logo + "Meridian" wordmark, laid out horizontally for headers and brand surfaces.
struct MeridianWordmark: View {
    var logoSize: CGFloat = 28

    var body: some View {
        HStack(spacing: 10) {
            MeridianLogo(size: logoSize)
            Text("Meridian")
                .font(.system(size: 22, weight: .black))   // titleLarge + Black
                .foregroundStyle(MeridianColors.onBackground)
        }
    }
}

#if DEBUG
#Preview("MeridianLogo", traits: .sizeThatFitsLayout) {
    VStack(spacing: 24) {
        MeridianLogo(size: 64)
        MeridianWordmark(logoSize: 32)
    }
    .padding()
    .background(MeridianColors.background)
}
#endif
